js/src/main/scala-2.10/scalajson.ast/JValue.scala

```diff
@@ -97,15 +102,9 @@ object JNumber {
   */
 // Due to a restriction in Scala 2.10, we cant override/replace the default apply method
 // generated by the compiler even when the constructor itself is marked private
-final class JNumber private[ast] (val value: String) extends JValue {
-
-  /**
-    * Javascript specification for numbers specify a [[scala.Double]], so this is the default export method to `Javascript`
-    *
-    * @param value
-    */
-  def this(value: Double) = this(value.toString)
-
+final class JNumber private[ast] (val value: String)(
+    private[ast] val constructedFlag: Int)
```

Comment:
Could `constructedFlag` be a smell that we are doing poor man's object orientation?
If we want lossless round trip from Double -> JNumber -> Double, and Json -> AST -> Json, would it make sense to split this class into `private[ast] JInt`, `JLong`, `JFloat`, `JDouble`, `JBigInt`, `JBigDecimial`, and `JStringNumber`?